sphinx/screens-detail/new-contact/new-contact/src/main/java/chat/sphinx/new_contact/ui/NewContactViewModel.kt

```diff
@@ -37,15 +37,27 @@ internal class NewContactViewModel @Inject constructor(
         NewContactViewState
         >(dispatchers, NewContactViewState.Idle)
 {
+
+    val args: NewContactFragmentArgs by savedStateHandle.navArgs()
+
+    init {
+        val pubKey = args.argPubKey?.toLightningNodePubKey()
+
+        viewStateContainer.updateViewState(
+            NewContactViewState.NewContactLink(
+                args.argFromAddFriend,
+                pubKey?.getPubKey(),
+                pubKey?.getRouteHint()
+            )
+        )
+    }
```

Comment:
This will always start the screen in the `NewContactLink` view state, whether one was passed via args or not. 

Imo, these passed args (if present) should ride a `SideEffect` to populate the fields upon screen initialization, and not utilize the `ViewState` properties.

Might need to fix the Fragment's `onSideEffectCollect` method and adjust the `NewContactSideEffect.FromScanner` to populate the route hint field, but it'd be best to utilize that instead of a `ViewState`.